pyro/infer/autoguide/guides.py

```diff
@@ -688,29 +742,24 @@ def laplace_approximation(self, *args, **kwargs):
             poutine.replay(self.model, trace=guide_trace)).get_trace(*args, **kwargs)
         loss = guide_trace.log_prob_sum() - model_trace.log_prob_sum()
 
-        loc = pyro.param("{}_loc".format(self.prefix))
-        H = hessian(loss, loc.unconstrained())
+        H = hessian(loss, self.loc)
         cov = H.inverse()
+        loc = self.loc
         scale_tril = cov.cholesky()
 
-        # calculate scale_tril from self.guide()
-        scale_tril_name = "{}_scale_tril".format(self.prefix)
-        pyro.param(scale_tril_name, scale_tril,
-                   constraint=constraints.lower_cholesky)
-        # force an update to scale_tril even if it already exists
-        pyro.get_param_store()[scale_tril_name] = scale_tril
-
         gaussian_guide = AutoMultivariateNormal(self.model, prefix=self.prefix)
         gaussian_guide._setup_prototype(*args, **kwargs)
+        # Set loc, scale_tril parameters as computed above.
+        gaussian_guide.loc.data = loc
+        gaussian_guide.scale_tril_unconstrained.data = transform_to(constraints.lower_cholesky).inv(scale_tril)
```

Comment:
@fritzo - Is this the best way to reinitialize `scale_tril`?

Reply:
Hmm, I would hope we could either
```py
gaussian_guide.scale_tril = scale_tril
```
or if necessary
```py
del gaussian_guide.scale_tril
gaussian_guide.scale_tril = scale_tril
```
Does either of those work?

Reply:
(Note this PR is the most complex use of `ConstraintDescriptor` yet, and we may need to fix bugs)

Reply:
This works now if we directly inherit from `ConstrainedModule` without the `del`s. :)